### PR TITLE
Adding datatypes

### PR DESCRIPTION
**Pull request by JonasAls89**

### CODE REVIEW

File: templates/customer.json [hunk 1]
```diff
@@ -1,135 +1,140 @@
 [
```
Comment (BaardBouvet) [inline templates/customer.json]: This is wrong, it should not contain this warning.

File: templates/customer.json [hunk 2]
```diff
@@ -1,135 +1,140 @@
...
         },
-        "type": "pipe"
+        "type": "dtl"
+      }
+    ],
+    "type": "pipe"
+  },
+  {
+    "_id": "{{@ system @}}-{{@ datatype @}}-event-processed",
+    "comment": "WARNING! This pipe is generated from the template of the '{{@ datatype @}}' datatype and changes will be silently ignored during collapse. For more information see the connectorpy README.",
```
Comment (BaardBouvet) [inline templates/customer.json]: Same here

File: templates/customer.json [hunk 3]
```diff
@@ -1,135 +1,140 @@
...
+        "datatype": "{{@ datatype @}}"
+      },
+      "system": "{{@ system @}}",
+      "type": "rest"
+    },
+    "type": "pipe"
+  },
+  {
+    "_id": "{{@ system @}}-{{@ datatype @}}-register-webhook",
+    "comment": "WARNING! This pipe is generated from the template of the '{{@ datatype @}}' datatype and changes will be silently ignored during collapse. For more information see the connectorpy README.",
```
Comment (BaardBouvet) [inline templates/customer.json]: And here

File: templates/customer.json [hunk 4]
```diff
@@ -1,135 +1,140 @@
...
-            "properties": {
-                "datatype": "customer"
-            },
-            "system": "{{@ system @}}",
-            "type": "rest"
+    "type": "pipe"
+  },
+  {
+    "_id": "{{@ system @}}-{{@ datatype @}}-collect",
+    "comment": "WARNING! This pipe is generated from the template of the '{{@ datatype @}}' datatype and changes will be silently ignored during collapse. For more information see the connectorpy README.",
```
Comment (BaardBouvet) [inline templates/customer.json]: And here

File: templates/projectcategory.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
+  "_id": "{{@ system @}}-projectcategory-collect",
+  "namespaced_identifiers": false,
+  "source": {
+    "operation": "list",
+    "properties": {
+      "datatype": "project/category"
```
Comment (BaardBouvet) [inline templates/projectcategory.json:7]: I would call this property "url_part" or something to not confuse it with our datatype name which is the one in the pipe id. (But I'm not entirely sure, so we can discuss this one.)

**Review by BaardBouvet — CHANGES_REQUESTED** [review]

File: templates/projectcategory.json (hunk 5 above)
Comment (JonasAls89) [reply templates/projectcategory.json:7]: I'll let this stay as is for now then. But I get your point too.

File: templates/order.json [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    },
+    "type": "pipe"
+  },
+  {
+    "_id": "{{@ system @}}-{{@ datatype @}}-all",
+    "type": "pipe",
+    "source": {
+      "type": "embedded",
+      "entities": [{
+        "start": "~t2023-01-01T00:00:00Z"
```
Comment (BaardBouvet) [inline templates/order.json]: Did you commit something wrong here? We can't hardcode a specific date + 2 months..

File: templates/order.json [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+            ["dict", "type", "{{@ datatype @}}", "query",
+              ["concat", "{{@ datatype @}}DateFrom=",
+                ["datetime-format", "%Y-%m-%d", "_S.start"], "&{{@ datatype @}}DateTo=",
+                ["datetime-format", "%Y-%m-%d",
+                  ["datetime-plus", "month", 2, "_S.start"]
+                ]
+              ]
+            ]
+          ],
+          ["add", "::operation", "get-raw"]
```
Comment (JonasAls89) [inline templates/order.json]: I get en Exection Error: "reason_why_stopped": "Entity or transform configuration specified an unknown operation 'get-raw'. Don't know what to do." Is get-raw supposed to be added in the system config as a route as well?

File: templates/projectcategory.json (hunk 5 above)
Comment (BaardBouvet) [reply templates/projectcategory.json:7]: I agree with you. :)

File: templates/system.json [hunk 8]
```diff
@@ -4,31 +4,29 @@
   "operations": {
     "delete": {
       "method": "DELETE",
-      "url": "{{ properties.datatype }}/{{ entity.id }}"
+      "url": "{{ properties.datatype }}/{{ properties.id }}"
```
Comment (BaardBouvet) [inline templates/system.json]: We need to switch this to entity, that's what the share pipe expects.

**Review by BaardBouvet — CHANGES_REQUESTED** [review]
We need to switch from properties to entity in the system, that's what the share pattern requires, so we need to use that in the lookup of orderlines as well.

File: templates/system.json (hunk 8 above)
Comment (JonasAls89) [reply templates/system.json]: implemented.

**Review by BaardBouvet — APPROVED** [review]

**Pull request merged**